### PR TITLE
[ALLUXIO-3241] Fix impersonation regex

### PR DESCRIPTION
**Pull request by gpang**
https://alluxio.atlassian.net/browse/ALLUXIO-3241

### CODE REVIEW

**Comment by gpang** [comment]
@madanadit

**Comment by madanadit** [comment]
How about `adit.madan`? Looks like use of the delimiter period is a valid user name. 

Ref:   http://manpages.ubuntu.com/manpages/bionic/en/man5/adduser.conf.5.html

VALID NAMES
              adduser  and addgroup enforce conformity to IEEE Std 1003.1-2001, which allows only
              the following characters to appear  in  group  and  user  names:  letters,  digits,
              underscores,  periods,  at signs (@) and dashes. The name may no start with a dash.
              The "$" sign is allowed at the end of usernames (to conform to samba).

              An additional check can be adjusted via the configuration parameter  NAME_REGEX  to
              enforce a local policy.

**Comment by gpang** [comment]
@madanadit I updated the regex (and test) to include `.`.

**Comment by madanadit** [comment]
Thanks for adding period. At sign @ is the only other character that is also valid?

File: core/common/src/test/java/alluxio/PropertyKeyTest.java [hunk 1]
```diff
@@ -270,15 +270,15 @@ public void isBuiltIn() {
 
   @Test
   public void impersonationRegex() {
-    String name = "user-name";
-
     // test groups
+    String name = "a-a.b-b.c-c.groups";
```
Comment (madanadit) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: Nit: include upper case, underscore and digits in the test case

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 2]
```diff
@@ -3609,9 +3609,9 @@ private Name() {} // prevent instantiation
   public enum Template {
     LOCALITY_TIER("alluxio.locality.%s", "alluxio\\.locality\\.(\\w+)"),
     MASTER_IMPERSONATION_GROUPS_OPTION("alluxio.master.security.impersonation.%s.groups",
-        "alluxio.master.security.impersonation.(\\w+).groups"),
+        "alluxio\\.master\\.security\\.impersonation\\.([a-zA-Z_0-9-\\.]+)\\.groups"),
```
Comment (apc999) [inline core/common/src/main/java/alluxio/PropertyKey.java]: why there is no underscore ?

Comment (gpang) [reply core/common/src/main/java/alluxio/PropertyKey.java]: underscore is there, after `Z`.

File: core/common/src/test/java/alluxio/PropertyKeyTest.java (hunk 1 above)
Comment (gpang) [reply core/common/src/test/java/alluxio/PropertyKeyTest.java]: added more characters.

**Comment by madanadit** [comment]
Jenkins, test this please

**Pull request merged**